Your_second_ASDF_converter.ipynb

```diff
@@ -0,0 +1,1142 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "2ae2873f",
+   "metadata": {},
+   "source": [
+    "Writing your second ASDF converter tutorial\n",
+    "===================================\n",
+    "\n",
+    "This tutorial is intended to include the use of ASDF schemas\n",
+    "for the purposes of validating the ASDF files generated and\n",
+    "read by this converter.\n",
+    "\n",
+    "What are ASDF schemas for?\n",
+    "----------------------------------\n",
+    "\n",
+    "As such, it represents a different level of sophistication and\n",
+    "consequently, more work. But it has the benefit of ensuring that\n",
+    "the files it tries to read are legal against the expectations\n",
+    "of the converter. For example, if someone hand edited an ASDF\n",
+    "file where elements of the file include your serialized object,\n",
+    "and they modify that object in such a way as to invalidate the\n",
+    "object, the schema mechanism provides a way of checking that\n",
+    "they haven't broken it. This makes it simple to help ensure that\n",
+    "a file that is causing problems isn't with the file itself.\n",
+    "\n",
+    "An aside: the schema helps eliminate many problems, but not all\n",
+    "necessarily. For example if one of the attributes of your object\n",
+    "requires that it be a prime number, there is no schema mechanism\n",
+    "to ensure it is a prime number, but it can ensure that the file\n",
```

Comment:
https://iluxonchik.github.io/regular-expression-check-if-number-is-prime/

Reply:
There is always a smartass.